### ISSUE TITLE
Moving average ignored value of nan -- exploding loss during training

### ISSUE THREAD

**Issue by rileypsmith** [dbolya/yolact#518]
**Steps to Reproduce**

1. Create custom dataset with annotations in the style of COCO annotations from ADE20K dataset (available online)
2. Register dataset in config.py as follows:
```
ade_20k_dataset = dataset_base.copy({
    'name': 'ADE20K Dataset',

    'train_images': './data/ADE20K',
    'train_info': './data/ade_dataset.json',

    'valid_images': './data/ADE20K',
    'valid_info': './data/ade_dataset.json',

    'has_gt': True,
    'class_names': ('car', 'bus', 'truck', 'airplane') # I'm only interested in these categories so I eliminated all other annotations
})
```
3. Point base config to this custom dataset:
```
yolact_base_config = coco_base_config.copy({
    'name': 'yolact_base',

    # Dataset stuff
    'dataset': ade_20k_dataset,
    'num_classes': len(coco2017_dataset.class_names) + 1,
    ...
})
```
3. Identify two available GPUS:
`export CUDA_VISIBLE_DEVICES=0,1`
4. Train:
`python train.py --config=yolact_base_config --batch_size=16`

**Current Behavior**
![yolact training issue](https://user-images.githubusercontent.com/56838352/92513383-0e5c4200-f1c5-11ea-96ba-b96e42a8358d.jpg)
YOLACT trains for a while and then eventually starts encountering nan values and the loss explodes. 

**Expected Behavior**
YOLACT should complete training with no issues.

**Other Comments**
I have already reviewed similar issues like [this one](https://github.com/dbolya/yolact/issues/222) and [this one](https://github.com/dbolya/yolact/issues/318). I have tried the recommended approaches (running on only 1 GPU, varying the batch size by trying different multiples of the number of GPUs I am using, and varying the learning rate and the learning rate warmup schedule). All of these approaches affect how soon the loss explodes, but in all cases the loss eventually explodes as it encounters NaN values. 

I did dig into the YOLACT code to try and pinpoint exactly what is causing the issue and I did. The losses 'B', 'C', and 'S' are all causing the issue but the loss 'M' is not throwing any NaN values. Specifically, the Warning is triggered by line 322 in train.py: `loss_avgs[k].add(losses[k].item())`. But this does not throw an issue in the case that the loss in question is loss 'M'. 

Also, it is worth noting that unlike the other questions similar to this one I am not seeing any inf values, just NaN. I am running the latest version of the code. Also, for reference, there are just over 4000 images in my dataset. I thought it might have something to do with overfitting but the loss seems quite normal up to the point where it spirals out of control. What is perhaps most perplexing about this to me is that the loss seems to spiral out of control BEFORE the first Warning is triggered. Notice that at iteration 630, the loss is rather normal. At iteration 640, it has already exploded but the first Warning does not come until after this point. I also thought that maybe I setup the dataset improperly but if that were the case I would expect it to complain immediately.

Any ideas? Is this a bug or some issue on my end? Thanks in advance for the help.

**Comment by Aykatia** [comment]
I have same problem, Did you solve this issue ?
